packages/gatsby-source-shopify/src/create-resolvers.ts

```diff
@@ -0,0 +1,20 @@
+import { CreateResolversArgs } from "gatsby"
+import { getGatsbyImageFieldConfig } from "gatsby-plugin-image/graphql-utils"
+
+import { makeResolveGatsbyImageData } from "./resolve-gatsby-image-data"
+
+export function createResolvers(
+  { createResolvers, cache }: CreateResolversArgs,
+  { downloadImages, typePrefix = `` }: IShopifyPluginOptions
+): void {
+  if (!downloadImages) {
+    createResolvers({
+      [`${typePrefix}ShopifyImage`]: {
```

Comment:
Previously, you could do something like this and then pass `gatsbyImageData` to `<GatsbyImage />` like this example from the official Shopify starter: https://github.com/gatsbyjs/gatsby-starter-shopify/blob/main/src/components/product-card.jsx#L58-L62

```graphql
query product {
  shopifyProduct {
    images {
      gatsbyImageData
    }
  }
}
```

It would be nice if could keep this same functionality here too though I'm not 100% sure how to pull that off without a bit more of a deep dive. I'm thinking something like this...

```graphql
query product {
  shopifyProduct {
    media {
      gatsbyImageData
    }
  }
}
```

Reply:
I dug into this a bit more and it looks like you can achieve this already by doing something like this. (The filter might not be explicitly necessary)

```graphql
query products  {
  allShopifyProduct(filter: {media: {elemMatch: {mediaContentType: {eq: IMAGE}}}}) {
    nodes {
      media {
        mediaContentType
        preview {
          image {
            gatsbyImageData # this can be passed to `<GatsbyImage />`
          }
        }
      }
    }
  }
}
```

Reply:
Because media is an interface type in this case (represents the types ExternalVideo, MediaImage, Video and Model3d) you could use a fragment like so:

```
query products  {
  allShopifyProduct {
    nodes {
      media {
        ... on ShopifyImage {
          image {
            gatsbyImageData
          }
        }
      }
    }
  }
}
```

Reply:
It seems like we could restore `images` field by doing what `media` field does plus filtering to only return `ShopifyImage` items from it? (so purely schema customization addition, no changes to sourcing bits)

This possibly could mean easier migration path (reading snippet it seems users would be forced to use inline fragments on `ShopifyImage`, which while works, does have discoverability problems). My question here is if re-introduction of it make sense? Would it be considered to be "bad practice" using `images` only and not all of the `media` types when building frontends?

Reply:
@byronlanehill ICYMI ☝️